cloud/client.go

```diff
@@ -1111,6 +1165,385 @@ func (c *client) DeleteSatellite(ctx context.Context, name, orgID string) error
+	} else if strings.Contains(idOrEmail, "@") {
+		req.UserEmail = idOrEmail
+	} else {
+		return errors.Errorf("%q does not appear to be an ID or email address")
+	}
+
+	status, body, err := c.doCall(ctx, http.MethodPost, u, withAuth(), withJSONBody(req))
+	if err != nil {
+		return err
+	}
+
+	if status != http.StatusCreated {
+		return errors.Errorf("failed to add member to project: %s", body)
+	}
+
+	return nil
+}
+
+// UpdateProjectMember updates an existing member with the new permission
+func (c *client) UpdateProjectMember(ctx context.Context, orgName, name, userID, permission string) error {
+	u := fmt.Sprintf("/api/v0/projects/%s/%s/%s", orgName, name, userID)
+
+	req := &secretsapi.AddProjectMemberRequest{
+		Permission: permission,
+		UserId:     userID,
+	}
+
+	status, body, err := c.doCall(ctx, http.MethodPut, u, withAuth(), withJSONBody(req))
+	if err != nil {
+		return err
+	}
+
+	if status != http.StatusOK {
+		return errors.Errorf("failed to update member: %s", body)
+	}
+
+	return nil
+}
+
+// ListProjectMembers will return all project members if the user has permission to do so.
+func (c *client) ListProjectMembers(ctx context.Context, orgName, name string) error {
+	u := fmt.Sprintf("/api/v0/projects/%s/%s", orgName, name)
+
+	status, body, err := c.doCall(ctx, http.MethodGet, u, withAuth())
+	if err != nil {
+		return err
+	}
+
+	if status != http.StatusOK {
+		return errors.Errorf("failed to list project members: %s", body)
+	}
+
+	return nil
+}
+
+// RemoveProjectMember will remove a member from a project.
+func (c *client) RemoveProjectMember(ctx context.Context, orgName, name, userID string) error {
+	u := fmt.Sprintf("/api/v0/projects/%s/%s/%s", orgName, name, userID)
+
+	status, body, err := c.doCall(ctx, http.MethodDelete, u, withAuth())
+	if err != nil {
+		return err
+	}
+
+	if status != http.StatusOK {
+		return errors.Errorf("failed to remove a project member: %s", body)
+	}
+
+	return nil
+}
+
+// ListSecrets returns a list of secrets base on the given path.
+func (c *client) ListSecrets(ctx context.Context, path string) ([]*Secret, error) {
+	if !strings.HasPrefix(path, "/") {
+		path = "/" + path
+	}
+	u := "/api/v1/secrets" + path
+
+	status, body, err := c.doCall(ctx, http.MethodGet, u, withAuth())
+	if err != nil {
+		return nil, err
+	}
+
+	if status != http.StatusOK {
+		return nil, errors.Errorf("failed to list secrets: %s", body)
+	}
+
+	var secrets []*Secret
+
+	res := &secretsapi.ListSecretsResponse{}
+	err = jsonpb.UnmarshalString(body, res)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, secret := range res.Secrets {
+		secrets = append(secrets, &Secret{
+			Path:       secret.Path,
+			Value:      secret.Value,
+			CreatedAt:  secret.CreatedAt.AsTime(),
+			ModifiedAt: secret.ModifiedAt.AsTime(),
+		})
+	}
+
+	return secrets, nil
+}
+
+// SetSecret adds or updates the given path and secret combination.
+func (c *client) SetSecret(ctx context.Context, path string, secret []byte) error {
```

Comment:
What's the difference between this new function and the existing one we have for setting secrets: https://github.com/earthly/earthly/blob/42839f02736cc1e486628915afef9b1304e78adb/cloud/client.go#L714-L730

Reply:
So @vladaionescu asked me to keep the old code around and add the new methods/commands in parallel until we're ready to cut over. The new `SetSecret` method is just for the new project-based secret update endpoint. Perhaps some sort of prefix would help clarify, but I don't think the older code will be around for too much longer.